HW_11/Task_1_2.js

```diff
@@ -0,0 +1,194 @@
+//    Для корретной работы создайте дополнительно приватный метод getEmployeeIndex(firstName), и используйте его в removeEmployee
+// 3. Добавьте метод getTotalSalary() в классе Company, который возвращает сумму всех зарплат сотрудников.
+
+// 4. Добавление валидации для полей сотрудника в сеттеры:
+//   Employee:
+//     firstName и lastName — строка от 2 до 50 символов, только латинские буквы.
+//     profession — строка, не может быть пустой, только латинские буквы и пробелы.
+//     salary — число, должно быть больше 0 и меньше 10000.
+// 5. Проверьте свой код:
+//     const emp1 = new Employee("John", "Doe", "Developer", 3000);
+//     const emp2 = new Employee("Jane", "Smith", "Manager", 5000);
+//     const emp3 = new Employee("Mark", "Brown", "Designer", 4000);
+
+//     const company = new Company("Tech Corp", "123-456", "Main Street");
+//     company.addEmployee(emp1);
+//     company.addEmployee(emp2);
+//     company.addEmployee(emp3);
+
+//     console.log(company.getTotalSalary()); // 12000
+//     console.log(company.findEmployeeByName("Jane")); // Employee { firstName: 'Jane', ... }
+//     company.removeEmployee("John");
+//     console.log(company.getEmployees()); // [Employee, Employee]
+
+
+
+
+class Employee {
+    #salary
+
+    constructor (firstName, lastName, profession, salary) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.profession = profession;
+        this.#salary = salary
+    }
+
+    get firstName() {
+        return this._firstName
+    }
+    get lastName() {
+        return this._lastName
+    }
+    get profession() {
+        return this._profession
+    }
+    get salary() {
+        return this.#salary
+    }
+
+    set firstName(firstName) {
+        return this._firstName = firstName;
+    }
+    set lastName(lastName) {
+        return this._lastName = lastName;
+    }
+    set profession(profession) {
+        return this._profession = profession;
+    }
+    set salary(salary) {
+        return this.#salary = salary;
+    }
+
+    getFullName() {
+        return `${this._firstName} ${this.lastName}`;
+    }
+    
+}
+
+    // const emp1 = new Employee("John", "Doe", "Developer", 3100);
+    // console.log(emp1.firstName); // "John"
+    // emp1.salary = 3100;
+    // console.log(emp1.salary); // 3100
+
+class Company {
+    #employees
+    
+    constructor (title, phone, address, employees = []) {
+        this._title = title;
+        this._phone = phone;
+        this._address = address;
+        this.#employees = employees;
+    }
+
+    get title() {
+    return this._title;
+    }
+    get phone() {
+    return this._phone;
+    }
+    get address() {
+    return this._address;
+    }
+
+    addEmployee(employee) {
+    if (employee instanceof Employee) {
+    this.#employees.push(employee)
+    } else {
+    throw new Error("Объект не находится в классе Employee");
+    }
+    }
```

Comment:
Проверили, если не подходит, вернули ошибку
```suggestion
    addEmployee(employee) {
    if (!(employee instanceof Employee)) {
      throw new Error("Объект не находится в классе Employee");
    } 
    this.#employees.push(employee)
    }
```